### ISSUE TITLE
Freezing when debugging a server application

### ISSUE THREAD

**Issue by testforstephen** [microsoft/vscode#137772]
_Originally reported at https://github.com/microsoft/vscode-java-pack/issues/820_

- VS Code Version: 1.62.3
- OS Version: Windows 10

Steps to Reproduce:
1. Open a VS Code window and maximize it, then open a server application. (Java sample project we use is https://github.com/spring-projects/spring-petclinic, and JS server application can also reproduce the issue.)
2. Click F5 to trigger debugging.
3.  Add a breakpoint in source code. (OwnerController.java#86)
4.  Open a browser and maximize it, and input the server url to access the server application. (Open localhost:8080, go to "FIND OWNERS" bar and click "Find Owner").
5. The server side breakpoint is hit, and you will see the VS Code icon on the status bar is blinking. However, when clicking on it to display the VS Code window, the breakpoint location is not highlighted and the VS Code window seems to be frozen. minimizing and maximizing VS Code can restore the debugger state.

See the screencast from original issue owner about this issue.
https://drive.google.com/file/d/1xIoENYCvdDDDBto_Lufmpco_eJH_9sld/view?usp=sharing

I only reproduce it with Windows, macOS works fine.

**Comment by nickzhums** [comment]
This bug was reported during a customer interview and he has constantly experienced this issue when debugging his application

**Comment by weinand** [comment]
Actually, the underlying problem seems to be #123257.

Could you please try whether the problem disappears if you run VS Code with the command line flag `--disable-features=CalculateNativeWinOcclusion`.

**Comment by nickzhums** [comment]
cc @Tospaa as FYI

**Comment by testforstephen** [comment]
@weinand thanks for taking a look at the issue.

> --disable-features=CalculateNativeWinOcclusion

Tried this workaround, it works fine. The freezing issue disappear.

> "debug.focusWindowOnBreak": false

It works with this workaround as well. But the VS Code logo won't blink any more if clicking on browser to trigger a breakpoint. The experience is a little downgrade.

**Comment by weinand** [comment]
@testforstephen thanks for verifying my workarounds.

Yes, the command line flag is the preferred workaround until the problem has been resolved upstream.

**Comment by Tospaa** [comment]
Hello @weinand

Thanks for your suggestions. `--disable-features=CalculateNativeWinOcclusion` worked well indeed, but is there a way to use this flag everytime I open a VS Code window?

I tried to add this line into the `~\.vscode\argv.json` file:
```json
"disable-features": "CalculateNativeWinOcclusion"
```
but it didn't work. Actually I did not expect this to work because it is clearly stated at the top of the file that 
> Only a subset of arguments is currently supported

I also tried to add `--disable-features=CalculateNativeWinOcclusion` flag into the `C:\Users\<user>\AppData\Local\Programs\Microsoft VS Code\bin\code.cmd` file as an argument. It works if I launch the VS Code from the terminal using `code .`, but it doesn't work when I launch the VS Code from the File Explorer context menu. I could also add the same flag to the `HKEY_CURRENT_USER\SOFTWARE\Classes\Directory\shell\VSCode\command` key from the regedit, but I'm pretty sure those will be overridden with the updates. Even if they're not, I'd be happier if I can set the flag from one place to everywhere.

Is this possible somehow? Thanks in advance for any help!
